services/brig/src/Brig/Data/Client.hs

```diff
@@ -187,10 +189,23 @@ updatePrekeys u c pks = do
         _ -> return False
 
 claimPrekey :: UserId -> ClientId -> AppIO (Maybe ClientPrekey)
-claimPrekey u c = withOptLock u c $ do
-  prekey <- retry x1 $ query1 userPrekeys (params Quorum (u, c))
-  case prekey of
-    Just (i, k) -> do
+claimPrekey u c =
+  ifM
+    (view randomPrekeys)
+    -- Use random prekey selection strategy
```

Comment:
You could make the `then` and `else` blocks functions, give them good names, and avoid having to write comments here.  :)

Reply:
That's a style question but I did think of that and thought that I like to have the "workhorse" functions not be punted down to a where clause, `withLocalLock` and `withOptLock` are fairly self documenting already, commenting is caring :)